### DISCUSSION TITLE
Import/Export Schema documented?

### DISCUSSION THREAD

**Discussion by paulbowler** [pocketbase/pocketbase#5569] [Q&A]
Is the structure of the JSON schema format documented anywhere, including all field types and options?

**Comment by lumosminima** [comment]
Schema of what? your collections? or how pocketbase abstracts sqlite data types? for your own collections you can see structure in export collections in admin-ui, for supported field types: [https://pocketbase.io/docs/collections/](https://pocketbase.io/docs/collections/) if you want to see what pocketbase generates in the sqlite db you can open the db file with something like DB Browser for sqlite and look at the schema.

**Reply by paulbowler** [reply]
A schema for describing the JSON document after exporting a database using the admin console. See https://json-schema.org.

**Reply by lumosminima** [reply]
Pocketbase's import/export json doesn't adhere to json-schema spec and is pocketbase specific so you would probably need to map to json-schema yourself see also #1154. If your goal which you haven't mentioned is to get type safety on client then you can use [pocketbase-typegen](https://github.com/patmood/pocketbase-typegen) to generate collection types and then use [valibot](https://valibot.dev/)/[zod](https://zod.dev/) et al. to validate on client.
